### ISSUE TITLE
Document dependencies for installing plugins on Fedora

### ISSUE THREAD

**Issue by surajssd** [projectatomic/adb-atomic-developer-bundle#450]
When I install plugins like `vagrant-service-manager` and `vagrant-sshfs` it fails on ruby dependencies so its not documented anywhere, what all dependencies I need to install.

**Comment by praveenkumar** [comment]
@surajssd Try to install below dependencies. I also faced same issue and I think we require to document it. Most of time it fails for `nokogiri`.

```
dnf groupinstall "Development Tools"
dnf install rpm-build
```

**Comment by LalatenduMohanty** [comment]
I think the underlying issue is how to make sure the install steps mentioned in the documentation always stay valid and quickly recognize if it does not works when some packaging changes in Fedora or CentOS makes the steps invalid.

**Comment by surajssd** [comment]
@praveenkumar this be documented in the docs so that folks know about it!

**Comment by praveenkumar** [comment]
@surajssd agreed but did above steps fix the issues?

**Comment by surajssd** [comment]
@praveenkumar along with the above steps I had to install `zlib-devel` then `vagrant-service-manager` was installed.

I tried it on Fedora 24 and vagrant 1.8.1

**Comment by LalatenduMohanty** [comment]
I found something similar too. On Fedora 24 I had to install below packages to install VSM. 

```
$ sudo dnf install @"Development Tools"
$ sudo dnf install ruby-devel redhat-rpm-config  zlib-devel
```

**Comment by thatdocslady** [comment]
Hi @surajssd I'll be documenting this in the install steps, but just to make sure, is this required only for F24?

**Comment by surajssd** [comment]
@thatdocslady I have not tested those steps on other Fedora Versions actually, I tried those only on Fedora24

**Comment by hferentschik** [comment]
Has anyone tried to boil this down to a verified set of required packages? There seems to be some inconsistencies in the descriptions above. This is what worked on Fedora 23 to install service-manager

```
 # Get Vagrant with libvirt support 
 $ sudo dnf install vagrant vagrant-libvirt 

 # Do install plugins Vagrant needs to be able to compile native extensions. Basic dev tools are needed
 $ sudo dnf install @"Development Tools"

 # Ruby headers for native extension compilation 
 $ sudo dnf install ruby-devel

 # This one bugs me. I think we should determine what's really needed. There seems to be some
 # dev tool missing. Installing 'rpm-build' seems not intuitive 
 $ sudo dnf install rpm-build

 # To compile native extensions for Nokogiri
 $ sudo dnf install zlib-devel
```

**Comment by hferentschik** [comment]
So this works:

```
Vagrant.configure(2) do |config|

  config.vm.box = "bento/fedora-23"

  config.vm.provision "shell", inline: <<-SHELL
    sudo dnf install -y vagrant vagrant-libvirt
    sudo dnf install -y @"Development Tools" rpm-build
    sudo dnf install -y ruby-devel zlib-devel
    vagrant plugin install vagrant-service-manager
  SHELL
end
```

Seems _rpm_build_ is really needed (not just one of the dependencies it also pulls in). No idea really what _rpm_build_ does and why it would not be part of 'Development Tools'. Another Fedora packaging mystery.

**Comment by thatdocslady** [comment]
@hferentschik @surajssd can you please review PR #483 and see if it covers the required steps?

Also, I understand that we now require vagrant-sshfs, can you please verify that the dependencies are also required for that component, and also write the steps required to install sshfs so that I can add it as an install step? Thanks!

**Comment by bexelbie** [comment]
Fixed in #483 
Closing.
